server/src/main/java/org/opensearch/search/aggregations/bucket/range/RangeAggregator.java

```diff
@@ -256,24 +262,28 @@ public RangeAggregator(
         SearchContext context,
         Aggregator parent,
         CardinalityUpperBound cardinality,
-        Map<String, Object> metadata
+        Map<String, Object> metadata,
+        ValuesSourceConfig config
     ) throws IOException {
-
         super(name, factories, context, parent, cardinality.multiply(ranges.length), metadata);
         assert valuesSource != null;
         this.valuesSource = valuesSource;
         this.format = format;
         this.keyed = keyed;
         this.rangeFactory = rangeFactory;
+        this.ranges = ranges; // already sorted by the range.from and range.to
 
-        this.ranges = ranges;
+        fastFilterContext = new FastFilterRewriteHelper.FastFilterContext(context);
+        fastFilterContext.setAggregationType(new FastFilterRewriteHelper.RangeAggregationType(config, ranges));
```

Comment:
This agg type appears to be set immediately after the context is created every time - can we pass as a ctor param and make it final in `FastFilterContext` ?

Reply:
Worked on trying to refactor this, the problem is in composite aggregator, the values source config is provided using another class `CompositeValuesSourceConfig`. I did a little to at least able to pass in aggregation type as a parameter in constructor. But would like to do the overhaul refactor in another PR.